pkg/virt-api/api_test.go

```diff
@@ -43,23 +49,157 @@ import (
 const namespaceKubevirt = "kubevirt"
 
 var _ = Describe("Virt-api", func() {
-	var tmpDir string
+	var app virtAPIApp
+	var tmpDir, keyFile, certFile, signingCertFile, clientCAFile string
+	var goodPemCertificate1, goodPemCertificate2, badPemCertificate string
 	var server *ghttp.Server
+	var backend *httptest.Server
+	var ctrl *gomock.Controller
+	var authorizorMock *rest.MockVirtApiAuthorizor
+	var filesystemMock *MockFilesystem
+	var fileMock *MockFile
+	var expectedValidatingWebhooks *admissionregistrationv1beta1.ValidatingWebhookConfiguration
+	var expectedMutatingWebhooks *admissionregistrationv1beta1.MutatingWebhookConfiguration
+	var restrictiveMode os.FileMode
 	subresourceAggregatedApiName := v1.SubresourceGroupVersion.Version + "." + v1.SubresourceGroupName
-
 	log.Log.SetIOWriter(GinkgoWriter)
 
-	app := virtAPIApp{namespace: namespaceKubevirt}
 	BeforeEach(func() {
+		app = virtAPIApp{namespace: namespaceKubevirt}
 		server = ghttp.NewServer()
+
+		backend = httptest.NewServer(nil)
 		tmpDir, err := ioutil.TempDir("", "api_tmp_dir")
 		Expect(err).ToNot(HaveOccurred())
 		app.virtCli, _ = kubecli.GetKubevirtClientFromFlags(server.URL(), "")
 		app.certsDirectory = tmpDir
+		keyFile = filepath.Join(app.certsDirectory, "/key.pem")
+		certFile = filepath.Join(app.certsDirectory, "/cert.pem")
+		clientCAFile = filepath.Join(app.certsDirectory, "/clientCA.crt")
+		signingCertFile = filepath.Join(app.certsDirectory, "/signingCert.pem")
+		restrictiveMode = 0600
+
 		config, err := clientcmd.BuildConfigFromFlags(server.URL(), "")
 		Expect(err).ToNot(HaveOccurred())
 		app.authorizor, err = rest.NewAuthorizorFromConfig(config)
 		Expect(err).ToNot(HaveOccurred())
+		ctrl = gomock.NewController(GinkgoT())
+		authorizorMock = rest.NewMockVirtApiAuthorizor(ctrl)
+		filesystemMock = NewMockFilesystem(ctrl)
+		fileMock = NewMockFile(ctrl)
+
+		// Reset go-restful
+		http.DefaultServeMux = new(http.ServeMux)
+		restful.DefaultContainer = restful.NewContainer()
+		restful.DefaultContainer.ServeMux = http.DefaultServeMux
+
+		expectedMutatingWebhooks = &admissionregistrationv1beta1.MutatingWebhookConfiguration{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "MutatingWebhookConfiguration",
+				APIVersion: "admissionregistration.k8s.io/v1beta1",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name: virtWebhookMutator,
+				Labels: map[string]string{
+					v1.AppLabel: virtWebhookMutator,
+				},
+			},
+			Webhooks: app.mutatingWebhooks(),
```

Comment:
we need something that validates that the mutatingWebhook that is generated matches expected output.

Reply:
This is very declarative code, no algorithmic complexities here, testing this woul be like copy paste the actual code in the unit test and maintain it, don't know if it worth it.